pandas/core/internals.py

```diff
@@ -455,17 +455,23 @@ def downcast(self, dtypes=None, mgr=None):
 
         return blocks
 
-    def astype(self, dtype, copy=False, raise_on_error=True, values=None,
-               **kwargs):
-        return self._astype(dtype, copy=copy, raise_on_error=raise_on_error,
-                            values=values, **kwargs)
+    def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
+        return self._astype(dtype, copy=copy, errors=errors, values=values,
+                            **kwargs)
 
-    def _astype(self, dtype, copy=False, raise_on_error=True, values=None,
+    def _astype(self, dtype, copy=False, errors='raise', values=None,
                 klass=None, mgr=None, **kwargs):
         """
         Coerce to the new type (if copy=True, return a new copy)
         raise on an except if raise == True
         """
+        errors_legal_values = ('raise', 'ignore')
+
+        if errors not in errors_legal_values:
+            invalid_arg = "Expected value of kwarg 'errors' to be one of %s. "\
+                "Supplied value is '%s'" % (', '.join("'%s'" % arg for arg in
```

Comment:
Will do. I'll change the output message to:

```Expected value of kwarg 'errors' to be one of ['raise', 'ignore']. Supplied value 'True'```

This will do away with the need for the messy formatting.